### PR TITLE
Implemented Student self registration

### PR DESCRIPTION
**Pull request by nahloo-obi**
As Product Owner, I want the system to only allow a code of P -Present to be persisted if self-registration process is in operation since students should not be able to use any other codes.

### CODE REVIEW

File: ams/views/module_lesson_attendance.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from ..models.User import User
+from http import HTTPStatus
+
+
+bp = Blueprint('module-lessons-attendance', __name__, url_prefix='/api/v1/module-lessons-attendance')
+
+
+@bp.route("/<int:module_lesson_id>", methods=["POST"])
+@jwt_required()
+def Student_self_Attendance_Registration(module_lesson_id ):
```
Comment (OCE1960) [inline ams/views/module_lesson_attendance.py:20]: please your function naming convention is not in compliance with pep8 standard.
find below link as a guide.
https://peps.python.org/pep-0008/#function-and-variable-names

File: ams/views/module_lesson_attendance.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        student = Student.query.filter_by(user_id=user.id).first()
+        moduleLesson = ModuleLesson.query.get_or_404(module_lesson_id)
+
+        
+        if checkin_code is not None:
+
+            """
+                Check if this particular student is enrolled for this particular module
+            """
+            moduleEnrollment = ModuleEnrollment.query.filter_by(module_id=moduleLesson.module_id, student_id=student.id).first()
```
Comment (OCE1960) [inline ams/views/module_lesson_attendance.py:35]: please your variable naming convention is not in compliance with pep8 standard.
find below link as a guide.
https://peps.python.org/pep-0008/#function-and-variable-names

File: ams/views/module_lesson_attendance.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            if moduleEnrollment:
+                """
+                Validation of the checkin Code
+                """
+                if moduleLesson.checking_code == checkin_code: 
+                    moduleLessonAttendance = ModuleLessonAttendance(student_id=student.id, module_lesson_id=moduleLesson.id, attendance_status='Present')
```
Comment (OCE1960) [inline ams/views/module_lesson_attendance.py:41]: please your variable naming convention is not in compliance with pep8 standard.
find below link as a guide.
https://peps.python.org/pep-0008/#function-and-variable-names

Comment (OCE1960) [inline ams/views/module_lesson_attendance.py:41]: with reference to this ticket.
https://trello.com/c/eO2sqnGe/25-codes-for-student-attendance
`attendance_status='Present'` should be updated to `attendance_status='P'`

File: tests/test_module_lesson_attendance.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+def test_student_self_attendance_registration(client, auth):
```
Comment (OCE1960) [inline tests/test_module_lesson_attendance.py:1]: Your test cases are not complete.
you didn't test instances where the actor successfully checkin etc.
this is where you exercise the entirety of your code use case.

just for the sake of this project.
just add an additional test where the student successfully checkin.

**Review by OCE1960 — CHANGES_REQUESTED** [review]
some pull request review feedback

File: tests/test_module_lesson_attendance.py (hunk 4 above)
Comment (asmaUsman) [reply tests/test_module_lesson_attendance.py:1]: @nahloo-obi please consider adding an additional test case where a student is able to successfully check in

File: ams/views/module_lesson_attendance.py (hunk 3 above)
Comment (asmaUsman) [reply ams/views/module_lesson_attendance.py:41]: there is may need to add staff-id...

Comment (asmaUsman) [reply ams/views/module_lesson_attendance.py:41]: @nahloo-obi please! Function names should be lowercase, with words separated by underscores as necessary to improve readability. Variable names follow the same convention as function names...

**Review by OCE1960 — APPROVED** [review]
looks ok...

**Pull request merged**